apiserver/applicationoffers/base.go

```diff
@@ -141,6 +138,20 @@ func (api *BaseAPI) applicationOffersFromModel(
 	return results, nil
 }
 
+// checkOfferAccess returns the level of access the authenticated user has to the offer,
+// so long as it is greater than the requested perm.
+func (api *BaseAPI) checkOfferAccess(backend Backend, offerName string, perm permission.Access) (permission.Access, error) {
```

Comment:
permission.NoAccess is a legitimate enum value for a permission. It's not the same as getting a string value back and no knowing if empty string means something concrete or just missing, which is where we tend to require the use of the additional bool. I'll land now to progress and we can discuss further - easy to add it in the next PR if needed.